api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -0,0 +1,313 @@
+            dsd.setSqlFile(sqlFile.getAbsolutePath());
+        }
+        else {
+            throw new RuntimeException("SQL file " + dataSetDescriptor.getConfig() + " not found");
+        }
+
+        return dsd;
+    }
+
+    public static List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition, ReportDescriptor reportDescriptor) {
+
+        List<ReportDesign> reportDesigns = new ArrayList<ReportDesign>();
+
+        // always do a default CSV design
+        if (reportDescriptor.getDesigns() == null || reportDescriptor.getDesigns().size() == 0) {
+            DesignDescriptor defaultDesignDecsriptor = new DesignDescriptor();
+            defaultDesignDecsriptor.setType("csv");
+            reportDescriptor.setDesigns(Collections.singletonList(defaultDesignDecsriptor));
+        }
+
+        for (DesignDescriptor designDescriptor : reportDescriptor.getDesigns()) {
+            ReportDesign design = null;
+            if (designDescriptor.getType().equalsIgnoreCase("csv")) {
+                design = constructCSVReportDesign(reportDefinition);
+            }
+            else if (designDescriptor.getType().equalsIgnoreCase("excel") || designDescriptor.getType().equalsIgnoreCase("xls")) {
+                design = constructXlsReportDesign(reportDefinition, reportDescriptor, designDescriptor);
+            }
+            else {
+                throw new RuntimeException("Unsupported report design type: " + designDescriptor.getType() + " for report " + reportDefinition.getName());
+            }
+
+            design.addPropertyValue(ReportDesignRenderer.FILENAME_BASE_PROPERTY, StringUtils.replace(reportDefinition.getName(), " ", ".").toLowerCase() + "." +
+                    "{{ formatDate request.reportDefinition.parameterMappings.startDate \"yyyyMMdd\" }}." +
+                    "{{ formatDate request.reportDefinition.parameterMappings.endDate \"yyyyMMdd\" }}." +
+                    "{{ formatDate request.evaluateStartDatetime \"yyyyMMdd\" }}." +
+                    "{{ formatDate request.evaluateStartDatetime \"HHmm\" }}");
+
+            if (designDescriptor.getProperties() != null) {
+                for (Map.Entry<String,String> property : designDescriptor.getProperties().entrySet()) {
+                    design.addPropertyValue(property.getKey(), property.getValue());
+                }
+            }
+
+            reportDesigns.add(design);
+        }
+
+        return reportDesigns;
+    }
+
+    public static ReportDesign constructCSVReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign design = new ReportDesign();
+        design.setName("reporting.csv");
+        design.setReportDefinition(reportDefinition);
+        design.setRendererType(CsvReportRenderer.class);
+        return design;
+    }
+
+    public static ReportDesign constructXlsReportDesign(ReportDefinition reportDefinition, ReportDescriptor reportDescriptor, DesignDescriptor designDescriptor) {
```

Comment:
New method to construct Xls report Design, fyi... I stole the ReportDesignResource stuff from MirebalaisReports... it seemed generic enough to me, but lmk if you feel otherwise.